lock-api/src/main/java/com/palantir/lock/cache/ValueCacheUpdater.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.cache;
+
+import java.util.Set;
+
+public interface ValueCacheUpdater {
```

Comment:
annoying that we need this, but I buy it.

Reply:
Although actually there might be a decomp where we pair the caches _and_ don't need this.